hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -2084,9 +2084,20 @@ case class MatrixRepartition(child: MatrixIR, n: Int, shuffle: Boolean) extends
   override def columnCount: Option[Int] = child.columnCount
 }
 
-case class MatrixUnionRows(children: IndexedSeq[MatrixIR]) extends MatrixIR {
-
-  require(children.length > 1)
+case class MatrixUnionRows(unmappedChildren: IndexedSeq[MatrixIR]) extends MatrixIR {
+  require(unmappedChildren.length > 1)
+
+  val children = unmappedChildren.map { c =>
```

Comment:
I don't like this - I don't think the `unmappedChildren` should be stored in the node. 

How about a static `MatrixUnionRows.unify(mts: IndexedSeq[MatrixIR]): MatrixUnionRows` that does the proper transformations?

This will also fix the failing test.

Reply:
and unify should be called in the pruner as well.

Reply:
I'm not sure that `unify` needs to be called in the pruner. I think it may only need to be called in `Interpret`. The tests pass when it's called in Interpret at least.